### PR TITLE
Courtney/testing/top100

### PR DESCRIPTION
**Pull request by Courtney-Lippman**
# Description
Cypress tests for the top 100 user flows were created and are all passing. The rogue semi-colon at the bottom of the top 100 page was deleted. The Official site on the details page was changed to a hyperlink so that the user can click on the link and go to the official page of the game.
# Fixes # (issue)
#8 
# Type of change

- [x]  Bug fix (non-breaking change which fixes an issue)
 
- [ ]  New feature (non-breaking change which adds functionality)

- [ ]  Breaking change (fix or feature that would cause existing functionality not to work as expected)

- [ ] This change requires a documentation update

-  [ ] Styling

- [ ] Refactor

- [x] Testing

# Where were the changes?

top100.cy.js, details.json, top100.json, top100.tsx, Details.tsx

# Checklist:

- [x]  My code follows the style guidelines of this project
- [x]  I have performed a self-review of my code
- [ ]  I have commented my code, particularly in hard-to-understand areas (if applicable)
- [ ]  I have made corresponding changes to the documentation (if applicable)
- [x]  My changes generate no new warnings
- [x] All of the existing tests are passing

# Contributors (paired programming):
@Adelle-Pitsas @Nagel29

### CODE REVIEW

File: cypress/e2e/top100.cy.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  beforeEach(() => {
+  cy.intercept('https://api.boardgameatlas.com/api/search?order_by=rank&ascending=false&limit=100&client_id=NO0Fq8pQcF', {
+    method:"GET",
+    fixture: "../fixtures/top100.json"
+  })
+    cy.visit('http://localhost:3000/')
+  })
+
+  it('displays a title', () => {
+    cy.get("#title").contains('Boardgame')
```
Comment (knpausch) [inline cypress/e2e/top100.cy.js:13]: great use targeting specific items by Id

File: cypress/e2e/top100.cy.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  })
+
+  it('displays all top100 games', () => {
+    cy.get('.gameCard-container').should('have.length', 2)
+    cy.get('.gameCard-container').eq(0).should('contain', "Root")
+      .and('contain', '$48.00')
+      .and('contain', '4.06 / 5')
+      .and('contain', '404 reviews')
+      .within(() => {
+        cy.get('img').should('have.attr', 'src', "https://s3-us-west-1.amazonaws.com/5cc.images/games/uploaded/1540147295104")
```
Comment (knpausch) [inline cypress/e2e/top100.cy.js:31]: thanks for being specific and ensuring the correct image is loaded

File: cypress/e2e/top100.cy.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        cy.get('img').should('have.attr', 'src', "https://cdn.shopify.com/s/files/1/0513/4077/1515/products/scythe-board-game.jpg?v=1611090922")
+        cy.get('button').eq(1).should('exist')
+      })
+  })
+
+  it('favorites a game and changes color when heart is clicked', () => {
+    cy.get('.heart-container').eq(0)
+    .within(() => {
+      cy.get('.heart').should('exist')
+      .and('have.css', 'fill', 'rgb(137, 137, 137)')
```
Comment (knpausch) [inline cypress/e2e/top100.cy.js:49]: smart move checking the color of the heart to determine if it has been clicked. we used this technique too

File: cypress/e2e/top100.cy.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      .and('contain', '4.22 / 5')
+      .and('contain', '753 reviews')
+      .within(() => {
+        cy.get('img').should('have.attr', 'src', "https://cdn.shopify.com/s/files/1/0513/4077/1515/products/scythe-board-game.jpg?v=1611090922")
+        cy.get('button').eq(1).should('exist')
+      })
+  })
+
+  it('Should be able to search for a specific game', () => {
+    cy.get('#search-input').type('Roo')
```
Comment (knpausch) [inline cypress/e2e/top100.cy.js]: I recommend verifying the value of the search box contains "Roo" after use types it in the element

**Review by knpausch — APPROVED** [review]
Great work team! The only thing I would add is verifying the value of the search bar to "Roo" after use types it in to ensure it is showing exactly what they typed

**Review by Adelle-Pitsas — COMMENTED** [review]
Pulled down on my machine and all tests and ui is looking good to me!

**Pull request merged**